### PR TITLE
fix: getting imagePullSecrets should fail only if not exist in pod spec and service account

### LINKED ISSUE #266: Scanner Job fail fs one imagePullSecrets is missing
**What steps did you take and what happened:**

If you have a deployment with one or more imagePullSecrets and one of these does not exists, the scan job fail. 

**What did you expect to happen:**

Exactly As the kubernetes pods controller, just ignore the missing secret. 

**Anything else you would like to add:**

To reproduce, create a deployment with two imagePullSecrets. One existant and valid, the second non-existant

```
spec:
  containers:
  - name: private-reg-container
    image: <your-private-image>
  imagePullSecrets:
  - name: validRegCredSecret
  - name: missingRegCredSecret
```
Check the trivy-operator logs. 

The pod is correctly launched ( up and running ) , but the scanner job fail. 

**Environment:**

- Trivy-Operator version 0.1.0
- Kubernetes version 1.21

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>
## Description
Getting imagePullSecrets should fail only if not exist in pod spec and service account

## Related issues
- Close #266 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: pkg/kube/secrets.go [hunk 1]
```diff
@@ -154,12 +155,13 @@ func (r *secretsReader) ListByLocalObjectReferences(ctx context.Context, refs []
 	for _, secretRef := range refs {
 		var secret corev1.Secret
 		err := r.client.Get(ctx, client.ObjectKey{Name: secretRef.Name, Namespace: ns}, &secret)
-		if err != nil {
+		if err != nil && !k8sapierror.IsNotFound(err) {
```
Comment (erikgb) [inline pkg/kube/secrets.go:158]: Wouldn't it be better to just `continue` if `k8sapierror.IsNotFound(err)`? Then you avoid protecting the `append`.

File: pkg/kube/secrets.go [hunk 2]
```diff
@@ -189,7 +191,9 @@ func (r *secretsReader) ListImagePullSecretsByPodSpec(ctx context.Context, spec
 	if err != nil {
 		return nil, err
 	}
-
+	if len(secrets) == 0 && len(serviceAccountSecrets) == 0 {
+		return nil, fmt.Errorf("failed to list secrtes by imagePullSecrets ref %v and service account %s", spec.ImagePullSecrets, serviceAccountName)
```
Comment (erikgb) [inline pkg/kube/secrets.go:195]:
```suggestion
		return nil, fmt.Errorf("failed to list secrets by imagePullSecrets ref %v and service account %s", spec.ImagePullSecrets, serviceAccountName)
```

File: pkg/kube/testdata/fixture/sa_with_image_pull_secret.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "name": "default-token-uudge"
+    }
+  ],
+  "imagePullSecrets": [
+    {
+      "name": "myregistrykey"
+    }
+  ]
+}
```
Comment (erikgb) [inline pkg/kube/testdata/fixture/sa_with_image_pull_secret.json:21]:
```suggestion
}

```

File: pkg/kube/testdata/fixture/sa_without_image_pull_secret.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "namespace": "default",
+    "resourceVersion": "390",
+    "uid": "c5af80da-d909-44be-851c-8f68278d59f5"
+  },
+  "secrets": [
+    {
+      "name": "default-token-uudge"
+    }
+  ]
+}
```
Comment (erikgb) [inline pkg/kube/testdata/fixture/sa_without_image_pull_secret.json:16]:
```suggestion
}

```

**Review by erikgb — CHANGES_REQUESTED** [review]
One suggestion and a couple of nits.

File: pkg/kube/secrets.go [hunk 5]
```diff
@@ -155,11 +156,12 @@ func (r *secretsReader) ListByLocalObjectReferences(ctx context.Context, refs []
 		var secret corev1.Secret
 		err := r.client.Get(ctx, client.ObjectKey{Name: secretRef.Name, Namespace: ns}, &secret)
 		if err != nil {
+			if k8sapierror.IsNotFound(err) {
+				continue
+			}
 			return nil, fmt.Errorf("getting secret by name: %s/%s: %w", ns, secretRef.Name, err)
 		}
-		secrets = append(secrets, secret)
```
Comment (erikgb) [inline pkg/kube/secrets.go:160]: This statement should be be deleted.

**Comment by erikgb** [comment]
@chen-keinan I don't understand the logic in `impactPullSecretDefineAndNoSecretsFound` and why we need it. This code needs cleanup IMO, and I have drafted some here: https://github.com/aquasecurity/trivy-operator/pull/276/files. But your tests added in this PR is good!

**Comment by erikgb** [comment]
> We want it to fail if imagePullSecrets ref define , but secret do not exist , at least one secret must exist

Ok, I see. Is that how Kubernetes works? Ref.

> Exactly As the kubernetes pods controller, just ignore the missing secret.

**Review by josedonizetti — APPROVED** [review]

**Pull request merged**